deepcell/training/model_reconstitution_test.py

```diff
@@ -0,0 +1,198 @@
+# test.py
+# Test driver for the ModelTrainer class.
+#
+# Right now, I'm conceiving of the ModelTrainer class as taking in the three
+# conceptual elements. The elements are listed here and each element is
+# followed by the named arguments that need to be passed to the ModelTrainer
+# class to fully specify that element.
+#
+# - A dataset
+# (specified by X_train, y_train, X_test, and y_test)
+# - Train and validation data generators [ asumed for now to be ImageFullyConvDataGenerators]
+# (specified by train_generator and validation_generator)
+# - A model [ assumed for now to be bn_feature_net_skip_2D]
+# (specified by model)
+#
+# For now, the creation of these three elements is handled in this script. They
+# are each created in separate functions, though, and those functions could be
+# easily moved over to the ModelTrainer class. As part of this, ModelTrainer
+# would need to have its  __init__ arguments updated to include all the input
+# arguments needed for creation of the element. Those decisions are all really
+# about how big we want the scope of ModelTrainer to be.
+
+import deepcell
+import argparse
+import json
+import hashlib
+import re
+
+import os
+import random
+import numpy as np
+import tensorflow as tf
+
+#import deepcell_toolbox
+
+class ModelComparison():
+    def __init__(self, metadata_file_path):
+        with open(metadata_file_path) as metadata_file:
+            self.source_metadata = json.load(metadata_file)
+            self.seed_value = 13
+
+    def _compare_deepcell_version(self):
+        assert self.source_metadata["deepcell_version"] == deepcell.__version__
+
+    def _load_data(self):
+        # take in dataset
+        filename = 'HeLa_S3.npz'
+        test_size = 0.1 # % of data saved as test
+        (self.X_train, self.y_train), (self.X_test, self.y_test) = deepcell.datasets.hela_s3.load_data(
+                                                                   filename,
+                                                                   test_size = test_size,
+                                                                   seed = self.seed_value)
```

Comment:
It's best to create all class attributes in `__init__`. Without initializing attributes, the class could crash if it calls methods out of order.